gitclone/checkout_method_simple.go

```diff
@@ -0,0 +1,140 @@
+package gitclone
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+//
+// checkoutNone
+type checkoutNone struct{}
+
+func (c checkoutNone) do(gitCmd git.Git, fetchOptions fetchOptions, fallback fallbackRetry) error {
+	return nil
+}
+
+// CommitParams are parameters to check out a given commit (In addition to the repository URL)
+type CommitParams struct {
+	Commit string
+}
+
+// NewCommitParams validates and returns a new CommitParams
+func NewCommitParams(commit string) (*CommitParams, error) {
+	if strings.TrimSpace(commit) == "" {
+		return nil, NewParameterValidationError("commit checkout strategy can not be used, no commit hash specified")
+	}
+
+	return &CommitParams{
+		Commit: commit,
+	}, nil
+}
+
+//
+// checkoutCommit
+type checkoutCommit struct {
+	params CommitParams
+}
+
+func (c checkoutCommit) Validate() error {
+	if strings.TrimSpace(c.params.Commit) == "" {
+		return errors.New("no commit hash specified")
+	}
+
+	return nil
+}
+
+func (c checkoutCommit) do(gitCmd git.Git, fetchOptions fetchOptions, fallback fallbackRetry) error {
+	// Fetch then checkout
+	// No branch specified for fetch
+	if err := fetch(gitCmd, fetchOptions, nil); err != nil {
+		return err
+	}
+
+	if err := checkoutWithCustomRetry(gitCmd, checkoutArg{arg: c.params.Commit}, fallback); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// BranchParams are parameters to check out a given branch (In addition to the repository URL)
+type BranchParams struct {
+	Branch string
+	Commit *string
```

Comment:
why does it have `commit` as property?
If a commit is given we should do a commit checkout,

Reply:
removed, nice catch